spec/helpers/http_mock/tapping.lua

```diff
@@ -0,0 +1,47 @@
+--- Tapping implemented with http_mock
+---@module spec.helpers.http_mock.tapping
+
+local http_mock = require "spec.helpers.http_mock"
+
+local tapping = {}
+
+--- create a new tapping route.
+-- @tparam string|number target the target host/port of the tapping route
+-- @treturn table the tapping route
+function tapping.new_tapping_route(target)
+  if tonumber(target) then
+    -- TODO: handle the resovler!
+    target = "http://127.0.0.1:" .. target
+  end
+
+  if not target:find("://") then
+    target = "http://" .. target
+  end
+
+  return {
+    ["/"] = {
+      directives = [[proxy_pass ]] .. target .. [[;]],
+    }
+  }
+end
+
+--- create a new http_mock.tapping instance with a tapping route.
+-- @tparam string|number target the target host/port of the tapping route
+-- @tparam[opt] table|string|number listens the listen directive of the mock server, defaults to a random available port
+-- @tparam[opt="servroot_tapping"] string prefix the prefix of the mock server
+-- @tparam[opt] table log_opts see `http_mock.new`
+-- @treturn http_mock.tapping a tapping instance
+-- @treturn number the port the mock server listens to
```

Comment:
this 2nd return value is wrong, seems a copy-paste error, please update that it is correct.

Reply:
hmmm seems it only has one return value, so please remove the second one.

Reply:
its returning the function return value and lua will have 2 return value in this case.

Reply:
(the called function returns 2 so the calling function have a list of 2 values to return)